pkg/discovery/servicehandler.go

```diff
@@ -162,17 +227,20 @@ func shouldDiscoverAPI(endpoint string, discoveryTags, ignoreTags, apiTags []str
 }
 
 // updateSpec Updates the spec endpoints based on the given type.
-func updateSpec(specType, endpointURI, authPolicy string, specContent []byte) ([]byte, error) {
+func updateSpec(specType string, endpointURI string, authPolicy string, configuration map[string]interface{}, specContent []byte) ([]byte, error) {
```

Comment:
The first three params are all strings, so the string type can be dropped after `specType` and `endpointURI`